cmd/user-list/main.go

```diff
@@ -1,94 +1,67 @@
 	"github.com/edulinq/autograder/internal/util"
 )
 
 var args struct {
 	config.ConfigArgs
+	cmd.CommonOptions
 
-	Emails []string `help:"Optional list of users to limit the results to (leave empty for all users)." arg:"" optional:""`
-	Course string   `help:"Only include user enrolled in this course."`
-	Table  bool     `help:"Output data to stdout as a TSV." default:"false"`
+	Table bool `help:"Output data as a TSV." default:"false"`
 }
 
 func main() {
 	kong.Parse(&args,
-		kong.Description("List users on the server (default) or from the specified course."),
+		kong.Description("List users on the server."),
 	)
 
 	err := config.HandleConfigArgs(args.ConfigArgs)
 	if err != nil {
 		log.Fatal("Could not load config options.", err)
 	}
 
-	db.MustOpen()
-	defer db.MustClose()
+	request := users.ListRequest{}
 
-	if args.Course == "" {
-		listServerUsers(args.Emails, args.Table)
-	} else {
-		listCourseUsers(args.Emails, args.Course, args.Table)
-	}
-}
-
-func listServerUsers(emails []string, table bool) {
-	users, err := db.GetServerUsers()
-	if err != nil {
-		log.Fatal("Failed to get server users.", err)
-	}
-
-	if len(emails) > 0 {
-		newUsers := make(map[string]*model.ServerUser, len(emails))
-		for _, email := range emails {
-			newUsers[email] = users[email]
-		}
-
-		users = newUsers
+	var printFunc cmd.CustomResponseFormatter = nil
+	if args.Table {
+		printFunc = listServerUsersTable
 	}
 
-	if table {
-		fmt.Println(strings.Join(model.SERVER_USER_ROW_COLUMNS, "\t"))
-		for _, user := range users {
-			fmt.Println(strings.Join(user.MustToRow(), "\t"))
-		}
-	} else {
-		fmt.Println(util.MustToJSONIndent(users))
-	}
+	cmd.MustHandleCMDRequestAndExitFull(`users/list`, request, users.ListResponse{}, args.CommonOptions, printFunc)
 }
 
-func listCourseUsers(emails []string, courseID string, table bool) {
-	course := db.MustGetCourse(courseID)
+func listServerUsersTable(response core.APIResponse) string {
+	var responseContent users.ListResponse
+	util.MustJSONFromString(util.MustToJSON(response.Content), &responseContent)
 
-	users, err := db.GetCourseUsers(course)
-	if err != nil {
-		log.Fatal("Failed to get course users.", err)
-	}
+	var serverUsersTable strings.Builder
 
-	if len(emails) > 0 {
-		newUsers := make(map[string]*model.CourseUser, len(emails))
-		for _, email := range emails {
-			newUsers[email] = users[email]
+	headers := []string{"email", "name", "server-role", "courses"}
+	serverUsersTable.WriteString(strings.Join(headers, "\t") + "\n")
+
+	for i, user := range responseContent.Users {
+		if i > 0 {
+			serverUsersTable.WriteString("\n")
 		}
 
-		users = newUsers
+		serverUsersTable.WriteString(user.Email)
+		serverUsersTable.WriteString("\t")
+		serverUsersTable.WriteString(user.Name)
+		serverUsersTable.WriteString("\t")
+		serverUsersTable.WriteString(user.Role.String())
+		serverUsersTable.WriteString("\t")
+		serverUsersTable.WriteString(util.MustToJSONIndent(user.Courses))
```

Comment:
This breaks the output.
You wanna have one entry per row.

Reply:
By one entry do you mean one WriteString per row?

Reply:
No.

The idea with `--table` is to write out a TSV file (as the help description says).
You currently write lines like:
```
11	12	13	14	15
21	22	{
    "23a": "23b"
}	24	25
31	32	33	34	35
```

Which, of course, is not proper TSV.